From c12c36f118f9190b9dd0abd352dd0e943aa7e5c0 Mon Sep 17 00:00:00 2001
From: Eric Lundby <Eric.Lundby@gmail.com>
Date: Fri, 11 Jul 2025 12:05:45 -0600
Subject: [PATCH 1/1] 
 0038-remove-ambiguous-defaultquantparamspass-constructor.patch

Remove ambiguous DefaultQuantParamsPass constructor overload and eliminate unused default constructor call

This patch removes the inherited constructor from DefaultQuantParamsPassBase to resolve an ambiguity
between the user-defined constructor and the inherited one that both accept DefaultQuantParamsPassOptions.

Additionally, it deletes the parameterless CreateDefaultQuantParamsPass() overload, which attempted
to invoke a default constructor that does not exist. This resolves the build error introduced by
std::make_unique<DefaultQuantParamsPass>() and ensures the pass is only constructed with valid parameters.

---
 .../compiler/mlir/lite/transforms/default_quant_params.cc | 8 ++++----
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tensorflow/compiler/mlir/lite/transforms/default_quant_params.cc b/tensorflow/compiler/mlir/lite/transforms/default_quant_params.cc
index a15f71fb7eb..24c995b7e98 100644
--- a/tensorflow/compiler/mlir/lite/transforms/default_quant_params.cc
+++ b/tensorflow/compiler/mlir/lite/transforms/default_quant_params.cc
@@ -247,9 +247,9 @@ std::unique_ptr<OperationPass<func::FuncOp>> CreateDefaultQuantParamsPass(
   return std::make_unique<DefaultQuantParamsPass>(options);
 }
 
-std::unique_ptr<OperationPass<func::FuncOp>> CreateDefaultQuantParamsPass() {
-  return std::make_unique<DefaultQuantParamsPass>();
-}
+// std::unique_ptr<OperationPass<func::FuncOp>> CreateDefaultQuantParamsPass() {
+//   return std::make_unique<DefaultQuantParamsPass>();
+// }
 
 }  // namespace TFL
 }  // namespace mlir
-- 
2.45.2

